File: Backend/Adapters/Services/AIAgents.Laboratory.API.Adapters/Models/Response/ToolSkillDTO.cs
using AIAgents.Laboratory.API.Adapters.Models.Base;

namespace AIAgents.Laboratory.API.Adapters.Models.Response;

/// <summary>
/// The tool skill dto model.
/// </summary>
/// <seealso cref="BaseModelDTO"/>
public sealed record ToolSkillDTO : BaseModelDTO
{
    /// <summary>
    /// The tool skill id guid.
    /// </summary>
    public string ToolSkillGuid { get; set; } = string.Empty;

    /// <summary>
    /// The tool skill display name.
    /// </summary>
    public string ToolSkillDisplayName { get; set; } = string.Empty;

    /// <summary>
    /// The tool skill technical name.
    /// </summary>
    public string ToolSkillTechnicalName { get; set; } = string.Empty;

    /// <summary>
    /// The tool skill mcp server url.
    /// </summary>
    public string ToolSkillMcpServerUrl { get; set; } = string.Empty;

    /// <summary>
    /// The list of associated agents containing the agent guid and the agent name.
    /// </summary>
    public IList<AssociatedAgentsSkillDataDTO> AssociatedAgents { get; set; } = [];
}

File: UI/web/models/response/tool-skill-dto.ts
import { AssociatedAgentsSkillDataDTO } from "./associated-agents-skill-data.dto";

export class ToolSkillDTO {
	toolSkillGuid: string = "";
	toolSkillDisplayName: string = "";
	toolSkillTechnicalName: string = "";
	toolSkillMcpServerUrl: string = "";
	associatedAgents: AssociatedAgentsSkillDataDTO[] = [];
	dateCreated: Date = new Date();
	createdBy: string = "";
	dateModified: Date = new Date();
	modifiedBy: string = "";
}
